catchup/peerSelector.go

```diff
@@ -268,19 +274,32 @@ func (hs *historicStats) push(value int, counter uint64, class peerClass) (avera
 	return bounded
 }
 
-// makePeerSelector creates a peerSelector, given a peersRetriever and peerClass array.
-func makePeerSelector(net peersRetriever, initialPeersClasses []peerClass) *peerSelector {
-	selector := &peerSelector{
+// MakePeerSelectorExported exports makePeerSelector function
+func MakePeerSelectorExported(net peersRetriever, initialPeersClasses []PeerClassExported) *PeerSelector {
```

Comment:
You need a function to give you a peer selector, but the public API should not allow setting the peer classes.

Reply:
Are you suggesting something like merging `createPeerSelector` from service.go into this public constructor? Essentially `func MakePeerSelectorExported(net peersRetriever, cfg config.Local, pipelinedFetch boolean) *PeerSelector`